### ISSUE TITLE
AbbreviationAsWordInName incorrectly reports constants in annotations

### ISSUE THREAD

**Issue by athielke** [checkstyle/checkstyle#2694]
The `AbbreviationAsWordInNameCheck` doesn't view attributes defined in annotations to be constants:

The `SampleAnnotation.java` source:
```java
public @interface SampleAnnotation {
    String NAME = "SampleAnnotation";
}
```

Checkstyle config:
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="AbbreviationAsWordInName">
            <property name="allowedAbbreviationLength" value="1"/>
        </module>
    </module>
</module>
```

Output:
```
$ java -jar checkstyle-6.13-all.jar -c config.xml SampleAnnotation.java
Starting audit...
/Users/athielke/Development/checkstyle-issue/SampleAnnotation.java:2: error: Abbreviation in name must contain no more than '1' capital letters.
Audit done.
Checkstyle ends with 1 errors.
```

Since the `ignoreFinal` and `ignoreStatic` default to `true`, I would expect this usage to not raise any errors.

**Comment by romani** [comment]
http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_DEF

**Comment by romani** [comment]
merged.
